### ISSUE TITLE
In-App Web Browser?

### ISSUE THREAD

**Issue by tomtheall** [ventuso/Fi-Test#71]
If user clicks a hyperlink within the app, will it open on that user's native web browser or a browser in the app?

I recommend we install web browser functionality within the Fi app (search, forward/back button, refresh, etc.). How difficult is this to do? I assume apple (safari) and Android SDKs (chrome) allow for this but let us know if this is too difficult or requires design help.

The in-app web browser should launch when user clicks a hyperlink. But we should also let the user launch a new web page and browse the web without ever leaving the app. Reinforces the idea that Fi is a the first stop for internet needs…can we install a "Launch Web Browser" text button on all Applet menus?

See attached screenshot for reference on where to place the "Launch Web Browser" button.

![applet-menu](https://cloud.githubusercontent.com/assets/11476339/7008386/c3a0a8b0-dc57-11e4-8994-304766cf42fa.jpg)

**Comment by TarenJP** [comment]
Not sure I agree with this.   Let's not make it to easy to leave the
merchant page.

Taren Patterson
President
Fi Interactive,  Inc
(972) 922-2665

**Comment by tomtheall** [comment]
I agree that users should stay with merchant page…if we DON'T have a web browser, user will click on merchant content/link and be taken away to another app (Chrome, Safari, etc) to that view web content…thus taking them away from the app and getting distracted by other web pages.

By having an in-app browser, we are launching new web pages within Fi, keeping users with us and back button away from returning to network content.

Are you saying you just disagree with encouraging a "new" browsing session (the Launch Web Browser button)? If so, that's completely understandable, and let's NOT add this button in.

But do you agree an in-app browser needs to be installed for other normal hyperlink instances? …just not encouraging the launching of a new blank web-page ("brought to you by City Tavern" or something…starting with merchant's home page as the default)
Tom

**Comment by baonguyenventuso** [comment]
@tomtheall @TarenJP 

I am waiting for your final decision on this. 

My answer is yes, we can launch an in-app browser like Facebook does to go to a Merchant Site for example. 

Regards,

**Comment by TarenJP** [comment]
If we can keep the HTML browser framed inside Fi's "wrapper"  then I think
thats ok.

Taren Patterson
President
Fi Interactive,  Inc
(972) 922-2665

**Comment by tomtheall** [comment]
Yes, add in-app browser that opens upon clicking a hyperlink.

- All hyperlinks should be automatically detected and appear as blue in a chat message post.
- Clicking on the link opens the in-app web browser.

**Comment by tomtheall** [comment]
In addition to this…
1) Merchant's Website URLs (on each network's 2nd panel info card) should be clickable also. Launches the web browser. 

2) Clicking the phone number should also use the OS phone capabilities to dial that number.

(These DO NOT need to be blue though…keep same grey color and formatting just as is…people will know to click this, especially with the blue icon next to it...using google maps app as reference.)
